functions/scheduleinstance/index.js

```diff
@@ -25,37 +25,47 @@ const compute = new Compute();
  *
  * Expects a PubSub message with JSON-formatted event data containing the
  * following attributes:
- *  zone - the GCP zone the instance is located in.
- *  instance - the name of the instance.
+ *  zone - the GCP zone the instances are located in.
+ *  label - the label of instances to start.
  *
  * @param {!object} event Cloud Function PubSub message event.
- * @param {!object} callback Cloud Function PubSub callback indicating completion.
+ * @param {!object} callback Cloud Function PubSub callback indicating
+ *  completion.
  */
 exports.startInstancePubSub = (event, callback) => {
   try {
     const pubsubMessage = event.data;
     const payload = _validatePayload(
       JSON.parse(Buffer.from(pubsubMessage.data, 'base64').toString())
     );
-    compute
-      .zone(payload.zone)
-      .vm(payload.instance)
-      .start()
-      .then(data => {
-        // Operation pending.
-        const operation = data[0];
-        return operation.promise();
-      })
-      .then(() => {
-        // Operation complete. Instance successfully started.
-        const message = 'Successfully started instance ' + payload.instance;
-        console.log(message);
-        callback(null, message);
-      })
-      .catch(err => {
-        console.log(err);
-        callback(err);
+    const options = {
+      filter: 'labels.' + payload.label,
```

Comment:
Ditto to my other comment - template strings, and compress to one line?

Reply:
Done.